connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SinkConnectorConfig.java

```diff
@@ -96,6 +99,26 @@ public static void validate(Map<String, String> props) {
             throw new ConfigException("Must configure one of " +
                 SinkTask.TOPICS_CONFIG + " or " + SinkTask.TOPICS_REGEX_CONFIG);
         }
+
+        if (hasDlqTopicConfig) {
+            String dlqTopic = props.get(DLQ_TOPIC_NAME_CONFIG).trim();
+            if (hasTopicsConfig) {
+                String[] topics = props.get(SinkTask.TOPICS_CONFIG).split(",");
+                Arrays.setAll(topics, i -> topics[i].trim());
+                if (Arrays.asList(topics).contains(dlqTopic)) {
```

Comment:
I think we can use a utility method provided by the `ConfigDef` class here:
```suggestion
                List<String> topics = (List<String>) ConfigDef.parseType(SinkTask.TOPICS_CONFIG, props.get(SinkTask.TOPICS_CONFIG), ConfigDef.Type.LIST);
                if (topics.contains(dlqTopic)) {
```

Reply:
I agree but we have to take into account the following:
* That source code is also in ``WorkerSinkTask`` class.
* ``ConfigDef.parseType`` does not trim the ``split``. I mean, the result of splitting "topic1,topic2          " is not the same that " topic1        ,topic2".

I could include the changes you comment in this class and also in the ``WorkerSinkTask`` to be aligned. On the other hand, I would also fix the ``trim`` thing in ``ConfigDef.parseType``(or maybe in another PR).

What do you think?

Reply:
I think `ConfigDef::parseType` does do the trimming: https://github.com/apache/kafka/blob/9bc96d54f8953d190a1fb6478a0656f049ee3b32/clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java#L705

Changing the `WorkerSinkTask` to align seems fine, although if we're reusing this logic all over the place it might make sense to just add a utility method like `parseTopicsList(Map<String, Object>)` to something like the `SinkConnectorConfig` class that grabs the value for `SinkTask.TOPICS_CONFIG`, passes it to `ConfigDef::parseType`, and returns the result. That way we can just use that whenever we need to parse a topics list from a sink connector config.

Reply:
``ConfigDef.parseType`` does a trim but for the whole string, not the splits. This would be the issue in case you have spaces in each splitted topic name.

Reply:
I think it does do trimming for the splits, based on the regex that's used for the splitting. I tried out a few local examples and they seemed to confirm this; parsing something like `" \t foo \t, \tbar\t , b a z "` returned the list `["foo", "bar", "b a z"]` for me.

Reply:
Sorry about that. You're totally right.

Reply:
Haha, no worries! Does adding this logic to a reusable place like the `SinkConnectorConfig` sound good to you?

Reply:
Sure!